### PR TITLE
feat: Add Hamming loss function to sklearn metrics in Ivy frontends

### PR DESCRIPTION
**Pull request by muzakkirhussain011**
### Changes Proposed:
This PR introduces the Hamming loss function, a standard metric for binary classification performance evaluation, to the sklearn metrics module of Ivy frontends.

### Implementation Details:
- The `hamming_loss` function is implemented to compute the proportion of incorrect predictions.
- It supports both weighted and unweighted calculations.
- The implementation is designed to be compatible with binary classification tasks.

### Benefits:
- The addition of the Hamming loss provides Ivy users with a crucial metric for binary classification model evaluation.
- It enhances the Ivy metrics module, offering a comprehensive suite of performance evaluation tools.

### Usage:
The `hamming_loss` function can be utilized by passing the true and predicted binary labels, with an optional `sample_weight` parameter for a weighted assessment.

### Example:

from ivy.functional.frontends.sklearn.metrics import hamming_loss

y_true = [1, 0, 1, 0]
y_pred = [0, 1, 1, 0]
print(hamming_loss(y_true, y_pred))

## Checklist

- [x] Did you add a function?
- [x] Did you add the tests?
- [x] Did you run your tests and are your tests passing?
- [x] Did pre-commit not fail on any check?
- [x] Did you follow the steps we provided?

### CODE REVIEW

**Comment by muzakkirhussain011** [comment]
Hi @Ishticode ,

I hope this message finds you well.

I've submitted a PR for the addition of the Hamming loss function to the sklearn metrics in Ivy frontends, which is now ready for your review. This PR has successfully passed all backend tests as well as the CI tests, ensuring the robustness and reliability of the implementation. I've attached a screenshot of the passed backend tests for your reference.

Given your expertise and previous experience reviewing and merging the PR for the F1 score, I believe your insights would be invaluable for this PR as well.

Your timely review and feedback would be greatly appreciated. Please let me know if there are any additional details or changes required.

Thank you for your attention and support.

Best regards,

hamming_loss

**Comment by muzakkirhussain011** [comment]
Hi @Ishticode ,

I’ve submitted a PR for the Hamming loss function in Ivy frontends, which has passed all tests and is ready for review. Your expertise, especially considering your experience with the F1 score PR, would be greatly valuable. I look forward to your timely feedback.

Best Regards

**Review by Ishticode — APPROVED** [review]
Thank you very much work done for Hamming Loss @muzakkirhussain011 
Looks good. Will merge today. Thank you very much for the contribution :)
